src/sinks/aws_kinesis_streams.rs

```diff
@@ -305,7 +320,64 @@ mod integration_tests {
 
         let timestamp = timestamp as f64 / 1000.0;
         let records = rt
-            .block_on(fetch_records(STREAM_NAME.into(), timestamp, region))
+            .block_on(fetch_records(stream.clone(), timestamp, region))
+            .unwrap();
+
+        let mut output_lines = records
+            .into_iter()
+            .map(|e| String::from_utf8(e.data).unwrap())
+            .collect::<Vec<_>>();
+
+        input_lines.sort();
+        output_lines.sort();
+        assert_eq!(output_lines, input_lines)
+    }
+
+    #[test]
+    fn kinesis_put_records_partitioned() {
+        let stream = gen_stream();
+
+        let region = Region::Custom {
+            name: "localstack".into(),
+            endpoint: "http://localhost:4568".into(),
+        };
+
+        ensure_stream(region.clone(), stream.clone());
+
+        let config = KinesisSinkConfig {
+            stream_name: stream.clone(),
+            region: RegionOrEndpoint::with_endpoint("http://localhost:4568".into()),
+            batch_size: Some(2),
+            partition_key_field: Some("{{partition_key}}".into()),
+            ..Default::default()
+        };
+
+        let mut rt = Runtime::new().unwrap();
+
+        let sink = KinesisService::new(config, Acker::Null).unwrap();
+
+        let timestamp = chrono::Utc::now().timestamp_millis();
+
+        let (mut input_lines, _) = random_lines_with_stream(100, 11);
+        let events = input_lines
+            .clone()
+            .into_iter()
+            .enumerate()
+            .map(|(i, line)| {
+                let mut e = Event::from(line);
+                e.as_mut_log()
+                    .insert_implicit("partition_key".into(), format!("part-{}", i % 2).into());
+                e
+            });
+
+        let pump = sink.send_all(iter_ok(events));
+        rt.block_on(pump).unwrap();
+
+        std::thread::sleep(std::time::Duration::from_secs(1));
+
+        let timestamp = timestamp as f64 / 1000.0;
+        let records = rt
+            .block_on(fetch_records(stream.clone(), timestamp, region))
```

Comment:
I don't see how this is testing that half of the events get written to one partition and half get written to another.